node/src/actors/chain_manager/mod.rs

```diff
@@ -1029,8 +1029,113 @@ impl ChainManager {
 
                     for co_tx in &block.txns.commit_txns {
                         let commit_pkh = co_tx.body.proof.proof.pkh();
+                        let dr_output = self
+                            .chain_state
+                            .data_request_pool
+                            .get_dr_output(&co_tx.body.dr_pointer)
+                            .unwrap();
+                        let collateral = dr_output.collateral;
+                        let commit_fee = dr_output.commit_and_reveal_fee;
+                        // Reset witnessing power
                         let _ =
                             stakes.reset_age(commit_pkh, Capability::Witnessing, current_epoch, 1);
+                        // Subtract collateral from staked balance
+                        let _ = stakes.reserve_collateral(commit_pkh, Wit::from(collateral));
+                        // Add commit reward
+                        let _ = stakes.add_reward(
+                            commit_pkh,
+                            Wit::from(commit_fee),
+                            block_epoch,
+                        );
+                    }
+
+                    // Add reveal rewards
+                    for re_tx in &block.txns.reveal_txns {
+                        let reveal_pkh = re_tx.body.pkh;
+                        let reveal_fee = self
+                            .chain_state
+                            .data_request_pool
+                            .get_dr_output(&re_tx.body.dr_pointer)
+                            .unwrap()
+                            .commit_and_reveal_fee;
+                        let _ = stakes.add_reward(
+                            reveal_pkh,
+                            Wit::from(reveal_fee),
```

Comment:
Shouldn't `reval_fee` rather be sent to `miner_pkh`?

Reply:
See above, fixed it.